### PR TITLE
Add cuda qualifiers to algebras

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR adds cuda qualifiers to algebra functions. 
Currently bunch of warning or erros messages come from traccc out of this

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -140,3 +141,9 @@ endif()
 
 # Set up the packaging of the project.
 include( algebra-plugins-packaging )
+
+# Set GPU-related options
+if(ALGEBRA_PLUGIN_BUILD_CUDA)
+  set( CMAKE_CUDA_STANDARD 17 CACHE STRING "The (CUDA) C++ standard to use" )
+  enable_language(CUDA)
+endif()
```
Comment (krasznaa) [inline CMakeLists.txt:149]: The `ALGEBRA_PLUGIN_BUILD_CUDA` variable is not used for anything at the moment... You didn't add any CUDA source files to the project. Making use of the modified headers "only" requires the clients of the `INTERFACE` libraries to set up their own builds with CUDA correctly.

I would not make any changes to this file...

**Comment by krasznaa** [comment]
The CI jobs seem to be queued. It just seems that GitHub is throttling us at the moment.

As we discussed today with @niermann999 and @paulgessinger, we will probably need to reduce the number of test configurations. Just to ease the load on our GitHub quota. (There is no actual quota for our CI jobs as far as I know, but we certainly get throttled when GitHub is busy with many jobs.)

**Review by krasznaa — APPROVED** [review]
Just to make it clear, I'm fine with the changes by now. :wink:

**Comment by niermann999** [comment]
Will merge when the CI is done

**Pull request merged**